### DISCUSSION TITLE
State Machine add-in is not available any more in the RoboApps ( Robot Studio )

### DISCUSSION THREAD

**Discussion by nkquynh98** [ros-industrial/abb_libegm#152] [Q&A] [answered]
Hi everyone,

I am setting up a new Irb1200 robot with IR5C controller. I want to control the robot remotely using ROS ABB. However, when I follow the instruction for installing the egm on abb, I cannot find the add-in State Machine anymore in the RobotApps section in Robot Studio. I remember that it used to be in there one year ago, but now it has been removed. 

Could anyone help me with this problem? Or does anyone still have the StateMachine add-in setup file ( Statemachine.rspak ) and could share it with me? 

Thank you all.

**Comment by TheodoroCardoso** [comment]
Hi @nkquynh98,

We're facing the exact same problem.
Have you managed to solve it by any chance?

I happened to notice all the StateMachine Add-In links on the README.md point to the [RobotApps Forum](https://forums.robotstudio.com/categories/robotapps/), where I couldn't find anything about that.

This [post on ROS Discourse](https://discourse.ros.org/t/best-practice-for-using-robotstudio-with-git-and-ros/8155/5) also popped up when I tried to google it.
Perhaps you could point us in the right direction here @jontje?

Thanks!!

**Comment by mohamedamrali1993** [comment]
Hi @nkquynh98 & @TheodoroCardoso,

I am having the same problem and I am brand new to RAPID and the entire ABB framework.

I found a [repo](https://github.com/ITI-THM/statemachine_add_in_rw7) in this [open issue](https://github.com/ros-industrial/abb_robot_driver/issues/39) that has the 2.0.0 version for RW7, I think it is intended for Omnicore controllers and my IRC5 has RW6.13 . Thought this might be helpful, would either of you try to install it and let me know? 

Unfortunately, I don't have RobotStudio (licensed version) yet, so I don't have a complete picture of the APP but one place I'd check is under the Add-Ins tab.

[Issue 77](https://github.com/ros-industrial/abb_libegm/issues/77) talks about plans to host it.

@jontje and @gavanderhoorn , sorry to @ you but we really need your help! The MachineState Add-in is no longer hosted on the ABB RobotApps website. We would super appreciate it if you can open-source it. Please and thank you!

**Comment by bryangd34** [comment]
Hello @mohamedamrali1993,

Were you able to find the Statemachine Add-in for your IRC5 controller? I too am looking for it.
Thank you.

**Comment by gavanderhoorn** [comment, marked as answer]
According to [RobotApps website has reached its end of life](https://forums.robotstudio.com/discussion/13528/robotapps-website-has-reached-its-end-of-life) on the RobotStudio Forums, ABB has decided to shut down the RobotApps site. This was posted in July 2022.

The post seems to imply only "official ABB add-ins" will be downloadable from within RobotStudio from then on.

It's very likely the way we published the StateMachine Add-in makes it "unofficial", hence the disappearance.

I'll see if this situation can be resolved properly, but for now, here is a copy of the add-in's installation files:

 - v1.0: [ABB.StateMachine-1.0.zip](https://github.com/ros-industrial/abb_libegm/files/10774911/ABB.StateMachine-1.0.zip) (thanks @kappa95)
 - v1.1: [ABB.StateMachine-1.1.zip](https://github.com/ros-industrial/abb_libegm/files/10754679/ABB.StateMachine-1.1.zip) (thanks @dave992)

I've not tested extensively, but you should be able to make RobotStudio "see" the add-in by extracting the `.zip` and moving the `ABB.StateMachine-1.1` directory to `C:\ProgramData\ABB\DistributionPackages` (note: `C:\ProgramData` is hidden by default in Windows, but you can access it by just entering it after using `win+r` to open the *Run* dialog and then pressing <kbd>enter</kbd>).

After a restart, RS should pick it up and it should be available for installation on your IRC5.

NOTE: this is the RW6.x compatible version of the add-in. There is *no* official RW7 version.

---

Edit: RW7 add-in has been available for some time now:

 - v2.0: [ABB.StateMachine-2.0.0.zip](https://github.com/user-attachments/files/23651533/ABB.StateMachine-2.0.0.zip)

**Reply by bryangd34** [reply]
Thank you very much @gavanderhoorn !

**Reply by traversaro** [reply]
Thanks a lot for handling this @gavanderhoorn and in particular for providing a copy of the StateMachine Add-in  !

**Reply by mohamedamrali1993** [reply]
Thanks so much, this is plenty of help!

**Reply by gavanderhoorn** [reply]
Added RW7 compatible v2.

Note: this is the `v2.0.0` distributed by ABB (downloaded using RobotStudio), not a community port.

**Comment by gavanderhoorn** [comment]
I've converted this to a discussion as it's not really an issue with the code in this repository.

`abb_libegm` is not the primary user of the state machine addin (that would be `abb_robot_driver`).

Refer to https://github.com/ros-industrial/abb_robot_driver/issues/83 for the issue on `abb_robot_driver`'s tracker.

**Reply by nkquynh98** [reply]
Thank you @gavanderhoorn

**Comment by kappa95** [comment]
If can help, I can upload the version 1.0 that is suitable for older robotware version: 
[ABB.StateMachine-1.0.zip](https://github.com/ros-industrial/abb_libegm/files/10774911/ABB.StateMachine-1.0.zip)

**Comment by dave992** [comment]
> See the Add-In's user manual ([1.0](https://robotapps.blob.core.windows.net/appreferences/docs/27e5bd15-b5ec-401d-986a-30c9d2934e97UserManual.pdf) or [1.1](https://robotapps.blob.core.windows.net/appreferences/docs/cd504500-80e2-4cb6-9419-c60ea4ad6d56UserManual.pdf)) for more details, as well as for install instructions for RobotWare systems. The manual can also be accessed by right-clicking on the Add-In in the Installed Packages list and selecting Documentation.

Probably related, the links to the manuals in the readme do not work for me anymore.

**Reply by gavanderhoorn** [reply]
Yes, this is essentially the same issue: the readmes were hosted on the same site. So with that gone, both the readme/landing page is inaccessible as well as the downloads.

**Comment by Qunnph** [comment]
why can't i use add-ins to get vacuum out from Robotstudio. i'm using 30 day trial license
